app/views/helpers/PDFViewHelper.scala

```diff
@@ -0,0 +1,309 @@
+            associatedCompanies
+          )}
+                ${pdfCorporationTaxHtml(messages("pdf.pageTotal3"), calculatorResult)}
+             ${pdfPageHtml(
+            nonTabCalculationResultsTable(
+              Seq(flatRate1, flatRate2),
+              associatedCompanies,
+              taxableProfit,
+              distributions,
+              config
+            ),
+            calculatorResult,
+            messages("pdf.pageTotal3")
+          )}
+             """)
+      case SingleResult(m: MarginalRate, _) =>
+        Html(s"""
+              ${pdfHeaderHtml(
+            messages("pdf.pageTotal3"),
+            pdfMetadata,
+            calculatorResult,
+            accountingPeriodForm,
+            taxableProfit,
+            distributions,
+            associatedCompanies
+          )}
+                ${pdfCorporationTaxHtml(messages("pdf.pageTotal3"), calculatorResult)}
+             ${pdfPageHtml(
+            nonTabCalculationResultsTable(Seq(m), associatedCompanies, taxableProfit, distributions, config),
+            calculatorResult,
+            messages("pdf.pageTotal3")
+          )}
+             """)
+      case DualResult(flatRate: FlatRate, m: MarginalRate, _) =>
+        Html(s"""
+              ${pdfHeaderHtml(
+            messages("pdf.pageTotal3"),
+            pdfMetadata,
+            calculatorResult,
+            accountingPeriodForm,
+            taxableProfit,
+            distributions,
+            associatedCompanies
+          )}
+                ${pdfCorporationTaxHtml(messages("pdf.pageTotal3"), calculatorResult)}
+             ${pdfPageHtml(
+            nonTabCalculationResultsTable(Seq(flatRate, m), associatedCompanies, taxableProfit, distributions, config),
+            calculatorResult,
+            messages("pdf.pageTotal3")
+          )}
+             """)
+
+      case DualResult(m: MarginalRate, flatRate: FlatRate, _) =>
+        Html(s"""
+                ${pdfHeaderHtml(
+            messages("pdf.pageTotal3"),
+            pdfMetadata,
+            calculatorResult,
+            accountingPeriodForm,
+            taxableProfit,
+            distributions,
+            associatedCompanies
+          )}
+                ${pdfCorporationTaxHtml(messages("pdf.pageTotal3"), calculatorResult)}
+                ${pdfPageHtml(
+            nonTabCalculationResultsTable(Seq(m, flatRate), associatedCompanies, taxableProfit, distributions, config),
+            calculatorResult,
+            messages("pdf.pageTotal3")
+          )}""")
+
+      case DualResult(m1: MarginalRate, m2: MarginalRate, _) =>
+        Html(s"""
+        ${pdfHeaderHtml(
+            messages("pdf.pageTotal4"),
+            pdfMetadata,
+            calculatorResult,
+            accountingPeriodForm,
+            taxableProfit,
+            distributions,
+            associatedCompanies
+          )}
+        ${pdfCorporationTaxHtml(messages("pdf.pageTotal4"), calculatorResult)}
+        ${pdfPageHtml(
```

Comment:
pdfPageHtml - could have a more meaningful name. It represents detailed calculation, so may be pdfDetailedCalculationHtml?